pkg/util/podstore/podstore.go

```diff
@@ -189,24 +191,29 @@ func (s *PodStore) Run(stopCh <-chan struct{}) {
 // worker runs a worker thread that just dequeues item from deleteQueue and
 // remove the item from prevPod.
 func (s *PodStore) worker() {
-	for s.processDeleteQueueItem() {
+	// Use the same object in each worker to delete from the indexer by key
+	// (UID), as there is no reason to allocate a new object for each call
+	// to processDeleteQueueItem.
+	podDeletionKey := &corev1.Pod{}
+	for s.processDeleteQueueItem(podDeletionKey) {
 	}
 }
 
-func (s *PodStore) processDeleteQueueItem() bool {
-	pod, quit := s.podsToDelete.Get()
+func (s *PodStore) processDeleteQueueItem(podDeletionKey *corev1.Pod) bool {
+	podUID, quit := s.podsToDelete.Get()
 	if quit {
 		return false
 	}
+	defer s.podsToDelete.Done(podUID)
+	pod := podDeletionKey
+	pod.UID = podUID
 	s.mutex.Lock()
 	defer s.mutex.Unlock()
-	err := s.pods.Delete(pod)
-	if err != nil {
-		klog.ErrorS(err, "Error when deleting Pod from deletion workqueue", "Pod", klog.KObj(pod))
-		return false
+	if err := s.pods.Delete(pod); err != nil {
+		klog.ErrorS(err, "Error when deleting Pod from store", "key", podUID)
+		return true
 	}
 	delete(s.timestampMap, pod.UID)
-	s.podsToDelete.Done(pod)
 	klog.V(4).InfoS("Removed Pod from Pod Store", "Pod", klog.KObj(pod))
```

Comment:
The log needs update as `pod` no longer has namespace and name.

Reply:
nice catch, fixed